test/hooks/use_welcomes_test.dart

```diff
@@ -0,0 +1,111 @@
+import 'dart:async' show Completer;
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:sloth/hooks/use_welcomes.dart';
+import 'package:sloth/src/rust/api/welcomes.dart';
+import 'package:sloth/src/rust/frb_generated.dart';
+import '../test_helpers.dart';
+
+Welcome _welcome(String id, BigInt createdAt) => Welcome(
+  id: id,
+  mlsGroupId: 'mls_$id',
+  nostrGroupId: 'nostr_$id',
+  groupName: 'Group $id',
+  groupDescription: '',
+  groupAdminPubkeys: const [],
+  groupRelays: const [],
+  welcomer: 'welcomer',
+  memberCount: 1,
+  state: WelcomeState.pending,
+  createdAt: createdAt,
+);
+
+class _MockApi implements RustLibApi {
+  List<Welcome> welcomes = [];
+  Completer<List<Welcome>>? completer;
+  final calls = <String>[];
+
+  @override
+  Future<List<Welcome>> crateApiWelcomesPendingWelcomes({
+    required String pubkey,
+  }) {
+    calls.add(pubkey);
+    if (completer != null) return completer!.future;
+    return Future.value(welcomes);
+  }
+
+  @override
+  dynamic noSuchMethod(Invocation invocation) => throw UnimplementedError();
+}
+
+final _api = _MockApi();
+
+late WelcomesResult Function() getResult;
+
+Future<void> _pump(WidgetTester tester, String pubkey) async {
+  getResult = await mountHook(tester, () => useWelcomes(pubkey));
+}
+
+void main() {
+  setUpAll(() => RustLib.initMock(api: _api));
+
+  setUp(() {
+    _api.welcomes = [];
+    _api.completer = null;
```

Comment:
🦥 : Completer still sounds like a weird/confusing name for me... but I guess that if it actually is a Completer object then is ok?